Taken from Debian's patchset to fix compilation with gcc-4.4 - JeR

--- libpqxx-2.6.9.orig/src/transaction_base.cxx
+++ libpqxx-2.6.9/src/transaction_base.cxx
@@ -18,6 +18,7 @@
  */
 #include "pqxx/compiler-internal.hxx"
 
+#include <cstring>
 #include <stdexcept>
 
 #include "pqxx/connection_base"
--- libpqxx-2.6.9.orig/src/binarystring.cxx
+++ libpqxx-2.6.9/src/binarystring.cxx
@@ -16,6 +16,7 @@
  */
 #include "pqxx/compiler-internal.hxx"
 
+#include <cstring>
 #include <new>
 #include <stdexcept>
 
--- libpqxx-2.6.9.orig/src/result.cxx
+++ libpqxx-2.6.9/src/result.cxx
@@ -17,6 +17,8 @@
  */
 #include "pqxx/compiler-internal.hxx"
 
+#include <cstdlib>
+#include <cstring>
 #include <stdexcept>
 
 #include "libpq-fe.h"
@@ -44,7 +46,7 @@
 pqxx::internal::result_data::~result_data() { PQclear(data); }
 
 
-void pqxx::internal::freemem_result_data(result_data *d) throw () { delete d; }
+void PQXX_LIBEXPORT pqxx::internal::freemem_result_data(result_data *d) throw () { delete d; }
 
 
 pqxx::result::result(pqxx::internal::pq::PGresult *rhs,
--- libpqxx-2.6.9.orig/src/cursor.cxx
+++ libpqxx-2.6.9/src/cursor.cxx
@@ -18,6 +18,7 @@
 #include "pqxx/compiler-internal.hxx"
 
 #include <cstdlib>
+#include <cstring>
 
 #include "pqxx/cursor"
 #include "pqxx/result"
--- libpqxx-2.6.9.orig/src/connection_base.cxx
+++ libpqxx-2.6.9/src/connection_base.cxx
@@ -21,6 +21,7 @@
 #include <cassert>
 #include <cerrno>
 #include <cstdio>
+#include <cstring>
 #include <ctime>
 #include <stdexcept>
 
--- libpqxx-2.6.9.orig/test/test003.cxx
+++ libpqxx-2.6.9/test/test003.cxx
@@ -1,4 +1,5 @@
 #include <cstdio>
+#include <cstdlib>
 #include <iostream>
 #include <stdexcept>
 
--- libpqxx-2.6.9.orig/test/test092.cxx
+++ libpqxx-2.6.9/test/test092.cxx
@@ -1,4 +1,5 @@
 #include <cassert>
+#include <cstring>
 #include <iostream>
 #include <list>
 
--- libpqxx-2.6.9.orig/test/test030.cxx
+++ libpqxx-2.6.9/test/test030.cxx
@@ -1,4 +1,5 @@
 #include <cstdio>
+#include <cstring>
 #include <iostream>
 
 #include <pqxx/connection>
--- libpqxx-2.6.9.orig/test/test022.cxx
+++ libpqxx-2.6.9/test/test022.cxx
@@ -1,4 +1,5 @@
 #include <cstdio>
+#include <cstdlib>
 #include <iostream>
 #include <stdexcept>
 
--- libpqxx-2.6.9.orig/test/test011.cxx
+++ libpqxx-2.6.9/test/test011.cxx
@@ -1,4 +1,5 @@
 #include <cstdio>
+#include <cstring>
 #include <iostream>
 
 #include <pqxx/connection>

